### PR TITLE
feat(quantic): added quantic-result-date component

### PR DESCRIPTION
**Pull request by nathanlb**
Added a component for displaying a result date field value in a result template context.

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticResultDate/quanticResultDate.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  /**
+   * (Optional) The label to display.
+   * @api
+   * @type {string}
+   * @defaultValue `none`
+   */
+  @api label;
+  /**
+   * A function used to format the displayed value.
+   * The default format uses the default format for your locale.
```
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticResultDate/quanticResultDate.js:35]: I find the current wording confusing. What would you think of something like this?

> By default, the date is formatted using the browser locale.

**Review by lbergeron — APPROVED** [review]

**Review by mmitiche — APPROVED** [review]
👌

File: packages/quantic/force-app/main/default/lwc/quanticResultDate/quanticResultDate.js (hunk 1 above)
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticResultDate/quanticResultDate.js:35]: Haha yeah it's pretty badly phrased as it is now. I clearly hadn't reread it. I'll take your suggestion thanks

File: packages/quantic/force-app/main/default/lwc/quanticResultDate/quanticResultDate.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+   * @api
+   * @type {Function}
+   * @param {number} value
+   * @returns {string}
+   */
+  @api formattingFunction = (value) =>
+    new Intl.DateTimeFormat(LOCALE).format(new Date(value));
+
+  error;
+  validated = false;
```
[the remark this answers is not in the document]
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticResultDate/quanticResultDate.js:45]: `validated` is assigned when declared so the type is inferred automatically. I think I would leave that one without a jsdoc. What do you think?

[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticResultDate/quanticResultDate.js:46]: All good 👌

**Review by ThibodeauJF — APPROVED** [review]

**Pull request merged**